src/machine/machinectl.c

```diff
@@ -1412,6 +1413,119 @@ static int shell_machine(int argc, char *argv[], void *userdata) {
         return process_forward(event, &forward, master, 0, machine);
 }
 
+static int get_settings_path(const char *name, char **ret_path) {
+        _cleanup_free_ char *file_alloc = NULL;
+        const char *file;
+
+        assert(name);
+        assert(ret_path);
+
+        if (!endswith(name, ".nspawn")) {
+                file_alloc = strjoin(name, ".nspawn");
+                if (!file_alloc)
+                        return log_oom();
+                file = file_alloc;
+        } else
+                file = name;
+
+        FOREACH_STRING(i, "/etc/systemd/nspawn", "/run/systemd/nspawn", "/var/lib/machines") {
+                _cleanup_free_ char *path = NULL;
+
+                path = path_join(i, file);
+                if (!path)
+                        return log_oom();
+
+                if (access(path, F_OK) >= 0) {
+                        *ret_path = TAKE_PTR(path);
+                        return 0;
+                }
+        }
+
+        return -ENOENT;
+}
+
+static int edit_settings(int argc, char *argv[], void *userdata) {
+        _cleanup_(edit_file_context_done) EditFileContext context = {
+                .remove_parent = false,
+        };
+        int r;
+
+        if (!on_tty())
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Cannot edit machine settings if not on a tty.");
+
+        if (arg_transport != BUS_TRANSPORT_LOCAL)
+                return log_error_errno(SYNTHETIC_ERRNO(EOPNOTSUPP),
+                                       "Edit is only supported on the host machine.");
+
+        r = mac_selinux_init();
+        if (r < 0)
+                return r;
+
+        STRV_FOREACH(machine, strv_skip(argv, 1)) {
+                _cleanup_free_ char *file = NULL, *path = NULL;
+
+                file = strjoin(*machine, ".nspawn");
+                if (!file)
+                        return log_oom();
+
+                r = get_settings_path(file, &path, NULL);
+                if (r < 0) {
+                        log_debug_errno(r,
+                                        "No existing settings file for machine '%s' found, creating.",
+                                        *machine);
```

Comment:
This is only when `-ENOENT`. And on `-ENOMEM`, we should not create new file:
```c
if (r == -ENOENT) {
        log_debug_errno(...);
        ...;
} else if (r < 0)
        return log_error_errno(r, "Failed to ...");
```

`"creating."` -> `"creating a new file."` is better?